sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/docker_client.py

```diff
@@ -23,9 +20,11 @@
 )
 from azure.ai.ml._local_endpoints.local_endpoint_mode import LocalEndpointMode
 from azure.ai.ml._local_endpoints.vscode_debug.vscode_client import VSCodeClient
+from azure.ai.ml._utils.utils import DockerProxy
 from azure.ai.ml._utils._logger_utils import initialize_logger_info
 from azure.ai.ml.constants._endpoint import LocalEndpointConstants
 
+docker = DockerProxy()
```

Comment:
Instead of having a wrapper. Shall we check if docker is installed when local_endpoints related sub package is loaded ? https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/__init__.py

Reply:
Unfortunately, that subpackage is loaded immediately rather than only when a local deployment is needed. I've just tested with a check in the `__init__.py` file you mentioned and reach a `ModuleNotFoundError` when importing `MLClient`. A change like this appears to require a relatively large refactor.